quora-service/src/main/java/com/upgrad/quora/service/business/GetAllQuestionsBusinessService.java

```diff
@@ -0,0 +1,39 @@
+package com.upgrad.quora.service.business;
+
+import com.upgrad.quora.service.dao.QuestionDao;
+import com.upgrad.quora.service.dao.UserDao;
+import com.upgrad.quora.service.entity.QuestionEntity;
+import com.upgrad.quora.service.entity.UserAuthTokenEntity;
+import com.upgrad.quora.service.exception.AuthorizationFailedException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+public class GetAllQuestionsBusinessService {
+    @Autowired
+    private UserDao userDao;
+
+    @Autowired
+    private QuestionDao questionDao;
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public UserAuthTokenEntity verifyAuthToken(final String accessToken) throws AuthorizationFailedException {
+        UserAuthTokenEntity userAuthTokenEntity = userDao.getUserAuthToken(accessToken);
+        if(userAuthTokenEntity == null){
+            throw new AuthorizationFailedException("ATHR-001","User has not signed in");
+        }else if(userAuthTokenEntity.getLogoutAt()!=null) {
+            throw new AuthorizationFailedException("ATHR-002","User is signed out.Sign in first to get all questions");
+        }
+        return userAuthTokenEntity;
+
+    }
+
+    public List<QuestionEntity> getAllQuestions()  {
```

Comment:
I think this must be a part of the service and naming this AuthorizationService just to handle the Authorization would be better.
As mentioned in the earlier PR by you, i would prefer keeping all services functionality in 'services' like in the above getAllQuestions would be better off in QuestionService